src/raft/raft.rs

```diff
@@ -1802,29 +1885,19 @@ impl<T: Storage> Raft<T> {
         self.pending_conf = false;
     }
 
-    pub fn set_progress(&mut self, id: u64, matched: u64, next_idx: u64) {
```

Comment:
Can you keep the method and add a new arg `is_learner`? Looks like there are a lot of places using the same pattern.